Cannot create mapping. Mapping definition … has unsupported parameters (Debezium -> Elasticsearch)

I am using debezium to capture mysql binlog to kafak and kafka-connect-es save data to ES. When my table contains NOT NULL TIMESTAMP field, the ES connect create mapping failed with following error.
```[2019-03-19 16:49:27,377] ERROR WorkerSinkTask{id=elasticsearch-mysql-sink-0} Task threw an uncaught and unrecoverable exception. Task is being killed and will not recover until manually restarted. (org.apache.kafka.connect.runtime.WorkerSinkTask:584)
org.apache.kafka.connect.errors.ConnectException: Cannot create mapping {"mysqldata":{"properties":{"id":{"type":"integer"},"update_time":{"type":"text","fields":{"keyword":{"type":"keyword","ignore_above":256}},"null_value":"1970-01-01T00:00:00Z"},"name":{"type":"text","fields":{"keyword":{"type":"keyword","ignore_above":256}}}}}} -- {"root_cause":[{"type":"mapper_parsing_exception","reason":"Mapping definition for [update_time] has unsupported parameters:  [null_value : 1970-01-01T00:00:00Z]"}],"type":"mapper_parsing_exception","reason":"Mapping definition for [update_time] has unsupported parameters:  [null_value : 1970-01-01T00:00:00Z]"}
        at io.confluent.connect.elasticsearch.jest.JestElasticsearchClient.createMapping(JestElasticsearchClient.java:254)
        at io.confluent.connect.elasticsearch.Mapping.createMapping(Mapping.java:66)
        at io.confluent.connect.elasticsearch.ElasticsearchWriter.write(ElasticsearchWriter.java:258)
        at io.confluent.connect.elasticsearch.ElasticsearchSinkTask.put(ElasticsearchSinkTask.java:162)
        at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:564)
        at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:322)
        at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:225)
        at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:193)
        at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:175)
        at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:219)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
[2019-03-19 16:49:27,413] INFO WorkerSinkTask{id=elasticsearch-mysql-sink-0} Committing offsets synchronously using sequence number 1: {mysqlsink.test.test2-1=OffsetAndMetadata{offset=1, metadata=''}, mysqlsink.test.test-1=OffsetAndMetadata{offset=0, metadata=''}, mysqlsink.test.test2-0=OffsetAndMetadata{offset=0, metadata=''}, mysqlsink.test.test-0=OffsetAndMetadata{offset=0, metadata=''}} (org.apache.kafka.connect.runtime.WorkerSinkTask:331)
[2019-03-19 16:49:27,418] ERROR WorkerSinkTask{id=elasticsearch-mysql-sink-0} Task threw an uncaught and unrecoverable exception (org.apache.kafka.connect.runtime.WorkerTask:177)
org.apache.kafka.connect.errors.ConnectException: Exiting WorkerSinkTask due to unrecoverable exception.
        at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:586)
        at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:322)
        at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:225)
        at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:193)
        at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:175)
        at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:219)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.kafka.connect.errors.ConnectException: Cannot create mapping {"mysqldata":{"properties":{"id":{"type":"integer"},"update_time":{"type":"text","fields":{"keyword":{"type":"keyword","ignore_above":256}},"null_value":"1970-01-01T00:00:00Z"},"name":{"type":"text","fields":{"keyword":{"type":"keyword","ignore_above":256}}}}}} -- {"root_cause":[{"type":"mapper_parsing_exception","reason":"Mapping definition for [update_time] has unsupported parameters:  [null_value : 1970-01-01T00:00:00Z]"}],"type":"mapper_parsing_exception","reason":"Mapping definition for [update_time] has unsupported parameters:  [null_value : 1970-01-01T00:00:00Z]"}
        at io.confluent.connect.elasticsearch.jest.JestElasticsearchClient.createMapping(JestElasticsearchClient.java:254)
        at io.confluent.connect.elasticsearch.Mapping.createMapping(Mapping.java:66)
        at io.confluent.connect.elasticsearch.ElasticsearchWriter.write(ElasticsearchWriter.java:258)
        at io.confluent.connect.elasticsearch.ElasticsearchSinkTask.put(ElasticsearchSinkTask.java:162)
        at org.apache.kafka.connect.runtime.WorkerSinkTask.deliverMessages(WorkerSinkTask.java:564)
        ... 10 more
```
Here is my table structure
```
CREATE TABLE `test` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `update_time` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  `name` varchar(255) NOT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
```
This is message which debezium sent to kafka
```
{
	"schema": {
		"type": "struct",
		"fields": [{
			"type": "int32",
			"optional": false,
			"field": "id"
		}, {
			"type": "string",
			"optional": false,
			"name": "io.debezium.time.ZonedTimestamp",
			"version": 1,
			"default": "1970-01-01T00:00:00Z",
			"field": "update_time"
		}, {
			"type": "string",
			"optional": false,
			"field": "name"
		}],
		"optional": true,
		"name": "mysqlsink.test.test.Value"
	},
	"payload": {
		"id": 8,
		"update_time": "2019-03-19T07:02:44Z",
		"name": "ddd"
	}
}
```

ES mapping json
```
{
	"mysqldata": {
		"properties": {
			"id": {
				"type": "integer"
			},
			"update_time": {
				"type": "text",
				"fields": {
					"keyword": {
						"type": "keyword",
						"ignore_above": 256
					}
				},
				"null_value": "1970-01-01T00:00:00Z"
			},
			"name": {
				"type": "text",
				"fields": {
					"keyword": {
						"type": "keyword",
						"ignore_above": 256
					}
				}
			}
		}
	}
}
```

I know the problem is debezium convert the mysql not null timestamp field to string and give the default value to 1970-01-01 00:00:00, so that ES connect put the "null_value": "1970-01-01T00:00:00Z" in a text property.

How can I parse the timestamp field in ES-connect to make the mapping created correctlly?

--- a/src/main/java/io/confluent/connect/elasticsearch/Mapping.java
+++ b/src/main/java/io/confluent/connect/elasticsearch/Mapping.java
@@ -197,13 +197,12 @@ public class Mapping {
           break;
         case ElasticsearchSinkConnectorConstants.STRING_TYPE:
         case ElasticsearchSinkConnectorConstants.TEXT_TYPE:
-          // IGNORE default values for text types as this is not supported by ES side.
+        case ElasticsearchSinkConnectorConstants.BINARY_TYPE:
+          // IGNORE default values for text and binary types as this is not supported by ES side.
           // see https://www.elastic.co/guide/en/elasticsearch/reference/current/text.html
+          // https://www.elastic.co/guide/en/elasticsearch/reference/current/binary.html
           // for more details.
           //defaultValueNode = null;
-          break;
-        case ElasticsearchSinkConnectorConstants.BINARY_TYPE:
-          defaultValueNode = JsonNodeFactory.instance.binaryNode(bytes(defaultValue));
           break;
         case ElasticsearchSinkConnectorConstants.BOOLEAN_TYPE:
           defaultValueNode = JsonNodeFactory.instance.booleanNode((boolean) defaultValue);

--- a/src/main/java/io/confluent/connect/elasticsearch/Mapping.java
+++ b/src/main/java/io/confluent/connect/elasticsearch/Mapping.java
@@ -197,7 +197,10 @@ public class Mapping {
           break;
         case ElasticsearchSinkConnectorConstants.STRING_TYPE:
         case ElasticsearchSinkConnectorConstants.TEXT_TYPE:
-          defaultValueNode = JsonNodeFactory.instance.textNode((String) defaultValue);
+          // IGNORE default values for text types as this is not supported by ES side.
+          // see https://www.elastic.co/guide/en/elasticsearch/reference/current/text.html
+          // for more details.
+          //defaultValueNode = null;
           break;
         case ElasticsearchSinkConnectorConstants.BINARY_TYPE:
           defaultValueNode = JsonNodeFactory.instance.binaryNode(bytes(defaultValue));

--- a/src/test/java/io/confluent/connect/elasticsearch/MappingTest.java
+++ b/src/test/java/io/confluent/connect/elasticsearch/MappingTest.java
@@ -16,6 +16,7 @@
 
 package io.confluent.connect.elasticsearch;
 
+import com.fasterxml.jackson.databind.JsonNode;
 import com.google.gson.JsonObject;
 
 import org.apache.kafka.connect.data.Date;
@@ -49,6 +50,29 @@ public class MappingTest extends ElasticsearchSinkTestBase {
     JsonObject mapping = Mapping.getMapping(client, INDEX, TYPE);
     assertNotNull(mapping);
     verifyMapping(schema, mapping);
+  }
+
+  @Test
+  public void testInferMapping() throws Exception {
+
+    Schema stringSchema = SchemaBuilder
+        .struct()
+        .name("record")
+        .field("foo", SchemaBuilder.string().defaultValue("0").build())
+        .build();
+    JsonNode stringMapping = Mapping.inferMapping(client, stringSchema);
+
+    assertNull(stringMapping.get("properties").get("foo").get("null_value"));
+
+    Schema intSchema =SchemaBuilder
+        .struct()
+        .name("record")
+        .field("foo", SchemaBuilder.int32().defaultValue(0).build())
+        .build();
+
+    JsonNode intMapping = Mapping.inferMapping(client, intSchema);
+    assertNotNull(intMapping.get("properties").get("foo").get("null_value"));
+    assertEquals(0, intMapping.get("properties").get("foo").get("null_value").asInt());
   }
 
   protected Schema createSchema() {
@@ -96,7 +120,6 @@ public class MappingTest extends ElasticsearchSinkTestBase {
   @SuppressWarnings("unchecked")
   private void verifyMapping(Schema schema, JsonObject mapping) throws Exception {
     String schemaName = schema.name();
-
     Object type = mapping.get("type");
     if (schemaName != null) {
       switch (schemaName) {

